rasa/cli/data.py

```diff
@@ -398,12 +399,18 @@ def _migrate_model_config(args: argparse.Namespace):
     domain = _get_domain(domain_file)
 
     rule_output_file = _get_rules_path(Path(args.out))
+    rules = []
 
     # TODO:
-    # 1. Migrate
-    # 2. Dump config
-    # 3. Dump rules
-    # 4. Add telemetry
+    # 1. Migrate FallbackPolicy
+    # 2. Add telemetry
+    # 3. Auto backup file?
+    rasa.core.config.migrate_mapping_policy_to_rules(model_configuration, domain, rules)
```

Comment:
optional: Not a blocker, but I'd prefer it if this returns the changed the things instead of changing it inplace. It's somewhat stylistic as it comes from a functional mindset, but I think it would also be clearer because the reader doesn't have to checkout `migrate_mapping_policy_to_rules` to see what it might change.

Reply:
I considered both approaches and what made the cut for me was because I didn't know how to simply make an update to the domain without modifying it in place. If there is a simple why of copying and making changes to it, I'm happy to update the code. It led to that decision because I couldn't have some objects modified in place and other objects copied.

Reply:
I think that returning it would still make the inplace change more obvious. To avoid the inplace thing:  You can copy the domain by converting it to a `Dict`, `deepcopy` it, and `Domain.load` it again

Reply:
sounds a bit painful - would we be ok adding the `__deepcopy__()` method to the Domain class?

```python
    def __deepcopy__(self, memo: Dict[int, Any]) -> "Domain":
        """
        This dunder method makes it possible to use deepcopy directly
        on a `Domain` instance.

        See https://docs.python.org/3/library/copy.html#copy.deepcopy
        for more implementation
        """
        domain_dict = self.as_dict()
        return self.__class__.from_dict(copy.deepcopy(domain_dict, memo))
```

Then it would make it possible to do `copy.deepcopy(domain)` easily, and make sure the domain is always deepcopied the right way

Reply:
yes please